helpers/tdhttp/test_api.go

```diff
@@ -568,6 +570,38 @@ func (t *TestAPI) CmpHeader(expectedHeader interface{}) *TestAPI {
 	return t
 }
 
+// CmpCookies tests the last request response cookies against
+// expectedCookies. expectedCookies can be a []*http.Cookie or a TestDeep
+// operator. Keep in mind that if it is a []*http.Cookie, it has to match
+// exactly the response cookies. Often only the presence of a
+// cookie key is needed:
+//
+//   ta := tdhttp.NewTestAPI(t, mux).
+//     PostJSON("/login", map[string]string{"name": "Bob", "password": "Sponge"}).
+//     CmdStatus(200).
+//     CmpCookie(td.SuperBagOf(td.Struct(&http.Cookie{Name: "cookie_session"}, nil)))
+//
+// It fails if no request has been sent yet.
+func (t *TestAPI) CmpCookies(expectedCookies interface{}) *TestAPI {
+	defer t.t.AnchorsPersistTemporarily()()
+
+	t.t.Helper()
+
+	if !t.checkRequestSent() {
+		t.cookiesFailed = true
+		return t
+	}
+
+	t.cookiesFailed = !t.t.RootName("Response.Cookie").
```

Comment:
I just realize that we should only set `t.*Failed` when they are true, to avoid ignoring an error when the same `Cmp*` method is chained:
```suggestion
	if !t.t.RootName("Response.Cookie").
		CmpLax(t.response.Result().Cookies(), expectedCookies, t.name+"cookies should match") {
		t.cookiesFailed = true
	}
```

Reply:
Fixed in #160